fastai/distributed.py

```diff
@@ -16,7 +20,7 @@ class DistributedTrainer(LearnerCallback):
     def on_train_begin(self, **kwargs):
         self.learn.model = DistributedDataParallel(self.learn.model, device_ids=[self.cuda_id],
                                                    output_device=self.cuda_id)
-        train_sampler = DistributedSampler(self.learn.data.train_ds)
+        train_sampler = DistributedSampler(self.learn.data.train_dl.dataset)
```

Comment:
Making sure we pass the right dataset to sampler

For language models, data.train_dl.dataset != data.train_ds.
data.train_dl.dataset is a LanguageModelPreLoader
data.train_ds is a LabelList